File: DuDoong-Common/src/main/java/band/gosrock/common/aop/ApiBlockingAspect.java
package band.gosrock.common.aop;


import band.gosrock.common.exception.DuDoongDynamicException;
import band.gosrock.common.helper.SpringEnvironmentHelper;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.aspectj.lang.ProceedingJoinPoint;
import org.aspectj.lang.annotation.Around;
import org.aspectj.lang.annotation.Aspect;
import org.springframework.stereotype.Component;

@Aspect
@Component
@Slf4j
@RequiredArgsConstructor
public class ApiBlockingAspect {

    private final SpringEnvironmentHelper springEnvironmentHelper;

    @Around("@annotation(band.gosrock.common.annotation.DevelopOnlyApi)")
    public Object checkApiAcceptingCondition(ProceedingJoinPoint joinPoint) throws Throwable {
        if (springEnvironmentHelper.isProdProfile()) {
            throw new DuDoongDynamicException(405, "Blocked Api", "not working api in production");
        }
        return joinPoint.proceed();
    }
}

File: DuDoong-Common/src/main/kotlin/band/gosrock/common/aop/ApiBlockingAspect.kt
package band.gosrock.common.aop

import band.gosrock.common.exception.DuDoongDynamicException
import band.gosrock.common.helper.SpringEnvironmentHelper
import org.aspectj.lang.ProceedingJoinPoint
import org.aspectj.lang.annotation.Around
import org.aspectj.lang.annotation.Aspect
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Aspect
@Component
class ApiBlockingAspect(
    private val springEnvironmentHelper: SpringEnvironmentHelper,
) {
    private val log = LoggerFactory.getLogger(javaClass)

    @Around("@annotation(band.gosrock.common.annotation.DevelopOnlyApi)")
    @Throws(Throwable::class)
    fun checkApiAcceptingCondition(joinPoint: ProceedingJoinPoint): Any? {
        if (springEnvironmentHelper.isProdProfile()) {
            throw DuDoongDynamicException(405, "Blocked Api", "not working api in production")
        }
        return joinPoint.proceed()
    }
}
